src/main/kotlin/org/wfanet/measurement/duchy/daemon/herald/LiquidLegionsV2Starter.kt

```diff
@@ -36,20 +33,14 @@ object LiquidLegionsV2Starter : ProtocolStarter {
   override suspend fun createComputation(
     computationStorageClient: ComputationsCoroutineStub,
     globalComputation: GlobalComputation,
-    duchyPosition: DuchyPosition,
+    protocolsSetupConfig: ProtocolsSetupConfig,
     blobStorageBucket: String
   ) {
+    protocolsSetupConfig.liquidLegionsV2.role
     val globalId: String = checkNotNull(globalComputation.key?.globalComputationId)
     val initialComputationDetails = ComputationDetails.newBuilder().apply {
-      // TODO: use fixed role for all computations in this duchy.
       liquidLegionsV2Builder.apply {
-        role = when (duchyPosition.role) {
-          DuchyRole.PRIMARY -> AGGREGATOR
-          else -> NON_AGGREGATOR
-        }
-        incomingNodeId = duchyPosition.prev
-        outgoingNodeId = duchyPosition.next
-        aggregatorNodeId = duchyPosition.primary
```

Comment:
I would imagine that being able to query the computations database for node configuration would be helpful when maintaining the system. Having it encoded in running code only will make it harder to discover bugs or latency associated with configuration of the computation.

That information is not available until the keys are generated at each duchy. But the Herald can still write the order to the database after all the keys have been retrieved from the Kingdom. I think we would just need to call UpdateComputationDetails in the ComputationsService to get that done.

https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/0f031f3c75813f78eecf52895ce2f94a37373784/src/main/proto/wfa/measurement/internal/duchy/computations_service.proto#L102

Reply:
That was my original plan. If we do that, two questions.

1. should the mill do that or the herald do that? this shouldn't matter, it just impacts where the "AggregatorId" flag would be.
2. Even we keep these three ids in the computationDetails, we still need to compute the order when the mill needs a partially combined key. So how about just store the entire ordered list here? i.e., a repeated String field.

Reply:
I think they both make sense in the Herald. It can be what sets the policy for executing a computation and the mill can be the thing to execute the computation. Having it set in the Herald is nice because it is a smaller surface area than the mills. Not as much worrying about binary skew and flag / feature rollout.

Replacing with a repeated string field seems fine to me. Could also have the partial key computed at the same time and stored by the herald so we don't need to look it up later. No strong opinions on this second option.

Reply:
sgtm.

Added the repeated filed to the proto, and add a TODO to do to use it later when dealing with per computation keys. Currently, the herald doesn't know the ElGamal keys yet, so it cannot set the duchy_order.